website/content/en/preview/getting-started/getting-started-with-karpenter/cloudformation.yaml

```diff
@@ -58,8 +66,39 @@ Resources:
               - ec2:DescribeSecurityGroups
               - ec2:DescribeSpotPriceHistory
               - ec2:DescribeSubnets
-              - pricing:GetProducts
-              - ssm:GetParameter
+            Condition:
+              StringEquals:
+                "ec2:Region": !Sub "${AWS::Region}"
+          - Effect: Allow
+            Resource: "*"
+            Action: pricing:GetProducts
+          - Effect: Allow
+            Resource: !Sub "arn:${AWS::Partition}:ssm:${AWS::Region}::parameter/aws/service/*"
+            Action: ssm:GetParameter
+          - Effect: Allow
+            Resource: !Sub "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:instance/*"
+            Action: ec2:TerminateInstances
+            Condition:
+              StringEquals:
+                "aws:ResourceTag/karpenter.sh/managed-by": !Sub "${ClusterName}"
```

Comment:
this won't work if someone updates infra before moving to Machines right?

Reply:
Yep, you're right. Discussed offline. Since CloudFormation doesn't natively allow you to do string interpolation with Map keys in YAML, we've transformed the `PolicyDocument` into a JSON string where we can do the interpolation. This allow us to do a StringEquals and StringLike condition like

```yaml
StringEquals:
  "aws:ResourceTag/kubernetes.io/cluster/${ClusterName}": "owned"
StringLike
  "aws:ResourceTag/karpenter.sh/provisioner-name": "*"
```